java/src/main/native/src/TableJni.cpp

```diff
@@ -900,15 +905,21 @@ JNIEXPORT long JNICALL Java_ai_rapids_cudf_Table_writeParquetFileBegin(
     for (size_t i = 0; i < meta_keys.size(); ++i) {
       metadata.user_data[meta_keys[i].get()] = meta_values[i].get();
     }
-
+    cudf::jni::native_jintArray precisions(env, j_precisions);
+    std::vector<uint8_t> v_precisions(
+        precisions.data(), precisions.data() + precisions.size());
+ 
     sink_info sink{output_path.get()};
     chunked_parquet_writer_options opts =
         chunked_parquet_writer_options::builder(sink)
             .nullable_metadata(&metadata)
             .compression(static_cast<compression_type>(j_compression))
             .stats_level(static_cast<statistics_freq>(j_stats_freq))
+            .int96_timestamps(static_cast<bool>(j_isInt96))
+            .decimal_precision(v_precisions)
             .build();
-    std::shared_ptr<pq_chunked_state> state = write_parquet_chunked_begin(opts);
+
+   std::shared_ptr<pq_chunked_state> state = write_parquet_chunked_begin(opts);
```

Comment:
Indentation is off here
```suggestion
    std::shared_ptr<pq_chunked_state> state = write_parquet_chunked_begin(opts);
```